statement.go

```diff
@@ -36,15 +73,56 @@ func (s *stmt) Close() error {
 	return nil
 }
 
-func (s *stmt) NumInput() int {
+// NumInput returns the number of placeholders in the statement.
+// It implements the driver.Stmt interface.
+func (s *Stmt) NumInput() int {
 	if s.closed {
 		panic("database/sql/driver: misuse of duckdb driver: NumInput after Close")
 	}
 	paramCount := C.duckdb_nparams(*s.stmt)
 	return int(paramCount)
 }
 
-func (s *stmt) bind(args []driver.NamedValue) error {
+// ParamName returns the name of the parameter at the given index (1-based).
+func (s *Stmt) ParamName(n int) string {
+	if s.closed {
+		panic("database/sql/driver: misuse of duckdb driver: ParamName after Close")
+	}
+
+	name := C.duckdb_parameter_name(*s.stmt, C.idx_t(n))
+	paramName := C.GoString(name)
+	C.duckdb_free(unsafe.Pointer(name))
+	return paramName
+}
+
+// ParamType returns the expected type of the parameter at the given index (1-based).
+func (s *Stmt) ParamType(n int) Type {
```

Comment:
Let's add a test for this error and ensure that we return `TYPE_INVALID` for out-of-bounds indexes.